src/clients.rs

```diff
@@ -204,9 +204,15 @@ pub async fn create_http_client(default_port: u16, service_config: &ServiceConfi
         Some(_) => "https",
         None => "http",
     };
-    let mut base_url = Url::parse(&format!("{}://{}", protocol, &service_config.hostname)).unwrap();
-    base_url.set_port(Some(port)).unwrap();
+
+    let mut base_url = Url::parse(&format!("{}://{}", protocol, &service_config.hostname))
+        .unwrap_or_else(|e| panic!("error parsing base url: {}", e));
+    base_url
+        .set_port(Some(port))
+        .unwrap_or_else(|_| panic!("error setting port: {} ", port));
+
```

Comment:
I just want to note that we simply `unwrap()` here as urls are validated prior to this during the config validation step. However, it doesn't hurt to add good panic messages here just in case.

e.g.
https://github.com/foundation-model-stack/fms-guardrails-orchestrator/blob/main/src/config.rs#L305